src/pixelator/graph/backends/protocol.py

```diff
@@ -65,33 +77,51 @@ def raw(self):
         ...
 
     @property
-    def vs(self):
+    def vs(self) -> VertexSequence:
         """Get a sequence of the vertices in the Graph instance."""
         ...
 
     @property
-    def es(self):
+    def es(self) -> EdgeSequence:
         """A sequence of the edges in the Graph instance."""
         ...
 
-    def vcount(self):
+    def vcount(self) -> int:
         """Get the total number of vertices in the Graph instance."""
         ...
 
-    def ecount(self):
+    def ecount(self) -> int:
         """Get the total number of edges in the Graph instance."""
         ...
 
     def get_adjacency_sparse(self) -> csr_matrix:
         """Get the sparse adjacency matrix."""
         ...
 
-    def connected_components(self):
+    def connected_components(self) -> VertexClustering:
         """Get the connected components in the Graph instance."""
         ...
 
-    def community_leiden(self, **kwargs):
-        """Run community detection using the Leiden algorithm."""
+    def community_leiden(
+        self,
+        objective_function: Literal["modularity", "cpm"] = "modularity",
+        n_iterations: int = 2,
+        beta: float = 0.01,
+        **kwargs,
+    ) -> VertexClustering:
+        """Run community detection using the Leiden algorithm.
+
+        Run community detection on the graph, using the Leiden algorithm.
+        As an example we use this to remove edges that jump between cells
+        due to chimeric PCR products.
+
+        :param objective_function: modularity or cpm (for Constant Potts Model (CPM))
```

Comment:
No, we haven't tested it. But for e.g. igraph it is the default choice.

However, we don't actually expose this on the cli. Only on the `Graph` api. How do you feel, do you think I should hard code the choice on a lower level, or should we keep it as an option here?

Reply:
Given that we have tested mainly "modularity" and that is the default one in `graspologic`, I would hard-code it and then, open it up when we need more functionality there. It will come.

But we can discuss this appart as well if you want.